charts/partials/templates/_topologyspreadconstraints.tpl

```diff
@@ -0,0 +1,23 @@
+{{ define "linkerd.topologySpreadConstraints" -}}
+{{- if .Values.enableTopologySpreadConstraints }}
+topologySpreadConstraints:
+- maxSkew: 1
+  topologyKey: failure-domain.beta.kubernetes.io/zone
+  whenUnsatisfiable: ScheduleAnyway
```

Comment:
@alpeb @adleong 

I'm inclined to think that if we switched this to `DoNotSchedule` the unexpected behavior may not be a problem.

For this to work we would need to ensure that the pods get scheduled after the condition _is_ satisfied again.

Thoughts?

Reply:
According to my testing, if that label is not there (which is the case in k3d for example), switching that will cause pods to not being scheduled at all. Removing that rule and leaving only the one for `kubernetes.io/hostname` still causes pods to not being allocated as expected during upgrades unfortunately.